### DISCUSSION TITLE
Verification Testing Failed: Chocolatey timed out waiting for the command to finish

### DISCUSSION THREAD

**Discussion by vladimiry** [chocolatey/home#96] [Q&A] [answered]
What are the steps I could perform in order for the package to get verified?

https://community.chocolatey.org/packages/electron-mail

The following lines from the [build log](https://gist.github.com/choco-bot/ba13f783e2402984e33bad2f140e3be9) look suspicious to me as the system could not install the app in 45 minutes:
```
2022-01-11 09:29:50,391 2172 [DEBUG] - Elevating permissions and running ["C:\Users\Administrator\AppData\Local\Temp\chocolatey\electron-mail\4.13.2\electron-mail-4.13.2-windows-x64-nsis-installer.exe" /S ]. This may take a while, depending on the statements.
2022-01-11 10:14:08,659 2172 [WARN ] - Chocolatey timed out waiting for the command to finish. The timeout 
 specified (or the default value) was '2700' seconds. Perhaps try a 
 higher `--execution-timeout`? See `choco -h` for details.
```

When I was creating the package I tested the installing by `choco install electron-mail -s .` and it worked well.

**Comment by TheCakeIsNaOH** [comment]
If a package does not work in the package verifier, you can request an exemption. This can be done by going to the package page and adding a review comment with the request and the reason that the package is not compatible with the verifier.

However, in this case, that log is likely indicating that the installer popped up an error/warning message, and just sat there waiting until choco timed out. But there is not enough information in the log to figure out why the installer is not finishing. I would suggest testing in the test environment so see what is going on. The test environment is very close to the verifier environment, so it generally can reproduce issues seen in the verifier.
https://github.com/chocolatey-community/chocolatey-test-environment

**Comment by TheCakeIsNaOH** [comment, marked as answer]
You probably need to take a dependency on the `vcredist140` package:
https://github.com/vladimiry/ElectronMail/blob/master/scripts/electron-builder/installer.nsh#L3-L5

**Comment by vladimiry** [comment]
> You probably need to take a dependency on the vcredist140 package:

Good finding. I completely missed the point that some Win systems come without this library. Despite being a runtime dependency there is a custom installing step embedded into nsis flow which I guess doesn't respect the `/S` flag (like automatic "yes / next" clicking). So going to declare `vcredist140` as a dependency in choco's terms and see what happens.

**Comment by pauby** [comment]
This is not a choco issue so I've moved it here.
